plugins/hls-cabal-plugin/test/Completer.hs

```diff
@@ -336,6 +338,27 @@ importCompleterTests =
           [Syntax.SecArgName (Syntax.Position row (col + 7)) (BS8.pack name)]
           []
 
+autogenFieldCompletionTests :: TestTree
+autogenFieldCompletionTests =
+  testGroup "Autogen Field Completer Tests"
+    [ testAutogenField "library"         "autogen-completion/autogen-completion.cabal" (Position 6 9)  ["autogen-modules:", "autogen-includes:"]
+    , testAutogenField "executable"      "autogen-completion/autogen-completion.cabal" (Position 11 9) ["autogen-modules:", "autogen-includes:"]
+    , testAutogenField "test-suite"      "autogen-completion/autogen-completion.cabal" (Position 16 9) ["autogen-modules:", "autogen-includes:"]
+    , testAutogenField "benchmark"       "autogen-completion/autogen-completion.cabal" (Position 21 9) ["autogen-modules:", "autogen-includes:"]
+    , testAutogenField "foreign-library" "autogen-completion/autogen-completion.cabal" (Position 26 9) ["autogen-includes:"]
+    , testAutogenField "common"          "autogen-completion/autogen-completion.cabal" (Position 29 9) ["autogen-includes:"]
```

Comment:
I don't see a reason not to check for `autogen-modules` here as well, I think we could add it.